src/org/javarosa/model/xform/CompactSerializingVisitor.java

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * A modified version of XFormSerializingVisitor that constructs
+ * a compact version of the form in a format that can be sent via SMS.
+ *
+ * @author Joel Dean
+ */
+public class CompactSerializingVisitor implements IInstanceSerializingVisitor {
+
+    private String resultText; // sms string to be returned
+    private String delimiter;
+
+    /**
+     * The serializer to be used in constructing XML for AnswerData elements
+     */
+    private IAnswerDataSerializer serializer;
+
+    public byte[] serializeInstance(FormInstance model, FormDef formDef) throws IOException {
+        return serializeInstance(model);
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.javarosa.core.model.utils.IInstanceSerializingVisitor#serializeInstance(org.javarosa.core.model.instance.FormInstance)
+     */
+    public byte[] serializeInstance(FormInstance model) throws IOException {
+        return this.serializeInstance(model, new XPathReference("/"));
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.javarosa.core.model.utils.IInstanceSerializingVisitor#serializeInstance(org.javarosa.core.model.instance.FormInstance, org.javarosa.core.model.IDataReference)
+     */
+    public byte[] serializeInstance(FormInstance model, IDataReference ref) throws IOException {
+
+        if (this.serializer == null) {
+            this.setAnswerDataSerializer(new XFormAnswerDataSerializer());
+        }
+        model.accept(this);
+        if (resultText != null) {
+            //Encode in UTF-16 by default, since it's the default for complex messages
+            return resultText.getBytes("UTF-16BE");
+        } else {
+            return null;
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.javarosa.core.model.utils.IInstanceSerializingVisitor#createSerializedPayload(org.javarosa.core.model.instance.FormInstance)
+     */
+    public IDataPayload createSerializedPayload(FormInstance model) throws IOException {
+        return createSerializedPayload(model, new XPathReference("/"));
+    }
+
+    public IDataPayload createSerializedPayload(FormInstance model, IDataReference ref)
+        throws IOException {
+
+        if (this.serializer == null) {
+            this.setAnswerDataSerializer(new XFormAnswerDataSerializer());
+        }
+        model.accept(this);
+        if (resultText != null) {
+            byte[] form = resultText.getBytes("UTF-8");
+            return new ByteArrayPayload(form, null, IDataPayload.PAYLOAD_TYPE_SMS);
+        } else {
+            return null;
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.javarosa.core.model.utils.ITreeVisitor#visit(org.javarosa.core.model
+     * .DataModelTree)
+     */
+    public void visit(FormInstance tree) {
+
+        TreeElement root = tree.getRoot();
+
+        delimiter = root.getAttributeValue(NAMESPACE_ODK, "delimiter");
+        if (delimiter == null) {
+            // for the spelling-impaired...
```

Comment:
If this was for a US/UK spelling difference or is for legacy reasons, then fine, but I think we shouldn't do this sort of thing. I mean where do you draw the line?

Reply:
Yeah we can take it out. It has been in there since the original SMSSerializingVisitor`.